peer/gossip.go

```diff
@@ -0,0 +1,141 @@
+package peer
+
+import (
+	"context"
+	"fmt"
+	"github.com/renproject/aw/dht"
+	"github.com/renproject/aw/transport"
+	"github.com/renproject/aw/wire"
+	"github.com/renproject/id"
+	"go.uber.org/zap"
+)
+
+type GossipFunc func(ctx context.Context, subnet id.Hash, contentID []byte) error
+
+// Gossiper returns a new set of callback functions and a gossip function that can be used to spread information throughout a network.
+func Gossiper(logger *zap.Logger, t *transport.Transport, contentResolver dht.ContentResolver, addressTable transport.Table, next Callbacks) (Callbacks, GossipFunc) {
+
+	gossip := func(ctx context.Context, subnet id.Hash, contentID []byte) error {
+		msg := wire.Msg{Version: wire.MsgVersion1, To: subnet, Type: wire.MsgTypePush, Data: contentID}
+
+		var chainedError error = nil
+		var receivers []id.Signatory
+		if subnet.Equal(&GlobalSubnet) {
+			receivers = addressTable.All()
```

Comment:
We should be selecting a random subset from all, not actually all. Remember, all is up to 10K peers! Consider introducing a new method on the table: `RandomSubnet(alpha int)`.